examples/api-tests/src/menus.spec.js

```diff
@@ -97,4 +108,63 @@ describe('Menus', function () {
         assert.notEqual(contextMenu.items.length, 0);
     });
 
+    it('rendering a new context menu should close the current', async function () {
+        const commandId = '__test_command_' + new Date();
+        const contextMenuPath = ['__test_first_context_menu_' + new Date()];
+        const contextMenuPath2 = ['__test_second_context_menu_' + new Date()];
+        toTearDown.push(commands.registerCommand({
+            id: commandId,
+            label: commandId
+        }, {
+            execute: () => { }
+        }));
+        toTearDown.push(menus.registerMenuAction(contextMenuPath, { commandId }));
+        toTearDown.push(menus.registerMenuAction(contextMenuPath2, { commandId }));
+
+        const access = contextMenuService.render({
+            anchor: { x: 0, y: 0 },
+            menuPath: contextMenuPath
+        });
+        toTearDown.push(access);
+        if (!(access instanceof BrowserContextMenuAccess)) {
+            assert.isTrue(access instanceof BrowserContextMenuAccess);
+            return;
+        }
```

Comment:
I must be blind, but I do not understand this.

Reply:
that's for type checking, ideally i will just do `assert.isTrue(access instanceof BrowserContextMenuAccess)`, but then ts does not proper type of `access`

It should be possible with https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions But our TS versions is below 3.7 :(

Someone should do https://github.com/eclipse-theia/theia/issues/7693